### PR TITLE
Task 1: Core-concepts

### PR DESCRIPTION
**Pull request by vposidailo**

### CODE REVIEW

File: MovieDB/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "babel-loader": "^8.0.5",
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-react": "^6.24.1",
+    "express": "^4.16.4",
+    "react": "^16.8.4",
+    "react-dom": "^16.8.4",
+    "react-scripts": "^2.1.8"
+  },
+  "browserslist": [
```
Comment (EvgeniyTT) [inline MovieDB/package.json:24]: HA! Cool thing, I didn't know about it.

File: MovieDB/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    ">0.2%",
+    "not dead",
+    "not ie <= 11",
+    "not op_mini all"
+  ]
+}
```
Comment (EvgeniyTT) [inline MovieDB/package.json:37]: Could you please also add https://eslint.org/  to dev-dependencies and setip it with `default`  or `airbnb` rules? Also would be good to add eslint plugin to your IDE.

File: MovieDB/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "forst-app",
```
Comment (EvgeniyTT) [inline MovieDB/package.json]: "forst-app" - typo?

File: MovieDB/src/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Search } from './components/search-component';
+import MovieTileItem from './components/movie-tile-item';
```
Comment (EvgeniyTT) [inline MovieDB/src/index.js:5]: Please be consistent. 
1. Imports. One component is imported as default, another not. Both approaches are fine in general, just choose the one and use it. More right might be to use `default` approach as far `component` - is what you expect to get from a component file.
2. Naming. `movie-tile-item`, `search-component`. If you want to add `component` postfix - it's fine, but then add it to all components and filenames. Or just remove from `search` :)

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+MoviewDB/node_modules/.bin/acorn
+MoviewDB/node_modules/
+MovieDB/node_modules/
```
Comment (EvgeniyTT) [inline .gitignore:1]: To many rows for a one `node_modules` folder :)  You probably need only `/node_modules` row and change file structure (more in a separate comment)

**Review by EvgeniyTT — CHANGES_REQUESTED** [review]
1. Let's change file structure a little bit. As far we'll have both `react` and 'node' parts in the same project makes sense to keep them them in the same repo, but as a 2 separate projects. So you want to have structure like

```
backend/
 -src/
 -package.json
 -.gitignore
frontend/
 -src/
   -components/
 -package.json
 -.gitignore
```

And you probably don't need `MoviewDB` folder

2. React (frontend) part looks good. Lets add express (backend) also.

File: MovieDB/package.json (hunk 3 above)
Comment (vposidailo) [reply MovieDB/package.json]: Will be present in next commit.

File: MovieDB/src/index.js (hunk 4 above)
Comment (vposidailo) [reply MovieDB/src/index.js:5]: Fixed!

File: MovieDB/package.json (hunk 3 above)
Comment (vposidailo) [reply MovieDB/package.json]: Fixed!

File: .gitignore (hunk 5 above)
Comment (vposidailo) [reply .gitignore:1]: Fixed.

File: MovieDB/src/index.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+// create element examples
+const header = React.createElement('div', null, React.createElement('h1', null, 'This is a header'));
+const middle = React.createElement('div', null, 'here should be search bar');
+const footer = React.createElement('div', null, 'Footer');
+
+// create function component examples
+const listOfMovie = (movie) => {
+    return <div><h4>{ movie.Title }</h4><p>{movie.Author}</p><p>{movie.Duration}</p></div>
```
Comment (EvgeniyTT) [inline MovieDB/src/index.js]: `movie.Title` - in js camelCase naming convention is used. `movie.title`, `movie.someProp`

File: MovieDB/src/index.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import  Search from './components/search';
+import MovieTileItem from './components/movie-tile-item';
+
+// create element examples
+const header = React.createElement('div', null, React.createElement('h1', null, 'This is a header'));
+const middle = React.createElement('div', null, 'here should be search bar');
+const footer = React.createElement('div', null, 'Footer');
+
+// create function component examples
+const listOfMovie = (movie) => {
```
Comment (EvgeniyTT) [inline MovieDB/src/index.js]: Naming is always a topic for a long discussion, so I  would just give my suggestions, but, of course, you can leave it as you wish: `listOfMovie ` -> `movies` or `movieList`

Comment (EvgeniyTT) [inline MovieDB/src/index.js]: > (movie)

 brackets are not required for a single argument, but still fine

File: MovieDB/src/index.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    return <div><h4>{ movie.Title }</h4><p>{movie.Author}</p><p>{movie.Duration}</p></div>
+}
+
+const firtMoview = {
```
Comment (EvgeniyTT) [inline MovieDB/src/index.js]: firtMoview -> firstMovie, but is there a second one? Might be just "movie"?

File: MovieDB/package.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "movie-app",
+  "version": "1.0.0",
+  "description": "Moview database",
```
Comment (EvgeniyTT) [inline MovieDB/package.json]: "Moview" typo

File: MovieDB/src/index.js (hunk 7 above)
Comment (vposidailo) [reply MovieDB/src/index.js]: Fixed. Renamed to movies.

Comment (vposidailo) [reply MovieDB/src/index.js]: Fixed. Removed unnecessary parenthesis.

File: MovieDB/src/index.js (hunk 8 above)
Comment (vposidailo) [reply MovieDB/src/index.js]: Fixed. As suggested named it as movie.

File: MovieDB/package.json (hunk 9 above)
Comment (vposidailo) [reply MovieDB/package.json]: Fixed. Sorry for the typo :)

File: MovieDB/package.json (hunk 2 above)
Comment (vposidailo) [reply MovieDB/package.json:37]: Added. By default use airbnb configurations.

**Comment by EvgeniyTT** [comment]
Hey! I see you've made a lot of changes, cool! What abot my comment to restructure thr project and add backend?

**Comment by vposidailo** [comment]
> Hey! I see you've made a lot of changes, cool! What abot my comment to restructure thr project and add backend?

Hey there. Yep. Also in progress with it. I want include it into next task. Task - 2 about webpack. What do you think?

**Comment by EvgeniyTT** [comment]
> > Hey! I see you've made a lot of changes, cool! What abot my comment to restructure thr project and add backend?
> 
> Hey there. Yep. Also in progress with it. I want include it into next task. Task - 2 about webpack. What do you think?

1. Let's don't mix tasks, we will quickly become puzzled what isn't done, what was done and where.
2. If you want to create backend in the separate PR - that's ok, just please rename "MoviewDB" folder to the "frontend" in this one.

**Comment by vposidailo** [comment]
> > > Hey! I see you've made a lot of changes, cool! What abot my comment to restructure thr project and add backend?
> > 
> > 
> > Hey there. Yep. Also in progress with it. I want include it into next task. Task - 2 about webpack. What do you think?
> 
> 1. Let's don't mix tasks, we will quickly become puzzled what isn't done, what was done and where.
> 2. If you want to create backend in the separate PR - that's ok, just please rename "MoviewDB" folder to the "frontend" in this one.

I am not mixing a task. Currently I have 2 separate branches where I can do my progress.  ok? Just want to set up all necessary env for task 2.

**Pull request closed without merging**